packages/jest-matchers/src/toThrowMatchers.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2014, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import type {MatchersObject} from './types';
+
+const {stringify, getType} = require('jest-matcher-utils');
+
+const matchers: MatchersObject = {
+  toThrowError(actual: Function, expected: string | Error | RegExp) {
+    let error;
+
+    try {
+      actual();
+    } catch (e) {
+      error = e;
+    }
+
+    if (typeof expected === 'string') {
+      return _toThrowMatchingStringOrRegexp(error, expected);
+    } else if (typeof expected === 'function') {
+      return _toThrowMatchingError(error, expected);
+    } else if (expected instanceof RegExp) {
+      return _toThrowMatchingStringOrRegexp(error, expected);
+    } else {
+      throw new Error(
+        'Unexpected argument passed. Expected to get ' +
+        `'string', 'Error class' or 'regexp'. Got: '${getType(expected)}' ` +
+        `'${stringify(expected)}'`,
+      );
+    }
+  },
+};
+
+const _toThrowMatchingStringOrRegexp = (
```

Comment:
We usually don't prefix things in Jets with `_`. It's easy to see what's public because of `module.exports` and in the future well be able to do `export const <foo>` so it will always be obvious.